components/triplestore/src/main/java/org/trellisldp/triplestore/TriplestoreResourceService.java

```diff
@@ -205,10 +208,22 @@ private Boolean createOrReplace(final IRI identifier, final Session session, fin
                                                 .findFirst().orElse(LDP.MemberSubject));
             });
         }
+
         // Set the parent relationship
-        if (container != null) {
+        if (nonNull(container)) {
             dataset.add(PreferServerManaged, identifier, DC.isPartOf, container);
         }
+
+        if (nonNull(binary)) {
```

Comment:
So, post this PR, do _all_ impls have to create these triples from the `Binary`, or is this just what `TriplestoreResourceService` does here because for `TRS`, _everything_ has to be triples?

Reply:
One goal of this PR is to explicitly provide flexibility to an implementation to do whatever it wants with that `Binary` data. The `TRS` puts the data into triples, but that's only because that's what a triplestore would do. I really wouldn't expect any other impl to do that -- and even if it did use triples, it could use whatever triples it wanted to.

Reply:
💯